projects/fw_102/src/fw_102.c

```diff
@@ -0,0 +1,38 @@
+#include <stdlib.h>
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+#include "wait.h"
+#define COUNTER_INCREMENT_MS 500;  // 500 milliseconds between incrementing counter A
```

Comment:
I see that you've defined 500 up here but you forgot to pass this into your calls to start the soft timer. The whole point of defining COUNTER_INCREMENT_MS is so we can use it later rather than just hardcoding 500 (ie we sort of use COUNTER_INCREMENT_MS like a const variable so we don't have to hardcode the numbers into our code).
If you aren't sure why we use #define versus a const variable, observe the following 
https://uwmidsun.atlassian.net/wiki/spaces/ELEC/pages/1591803936/C+Reference+for+Midnight+Sun+Firmware+Team